astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -2525,7 +2526,12 @@ def instantiate_class(self) -> bases.Instance:
             pass
         return bases.Instance(self)
 
-    def getattr(self, name, context=None, class_context=True):
+    def getattr(
+        self,
+        name: str,
+        context: InferenceContext | None = None,
+        class_context: bool = True,
+    ) -> list[NodeNG]:
```

Comment:
Are you sure about this? I belive `locals` is actually `SuccesfulInferenceResult` but typed incorrectly currently...

Reply:
See comment in other issue. This is not always the case and I think the typing should be updated here.